linkml/generators/pydanticgen/templates/validator.py.jinja

```diff
@@ -3,9 +3,9 @@
         pattern=re.compile(r"{{pattern}}")
         if isinstance(v,list):
             for element in v:
-                if not pattern.match(element):
+                if isinstance(v, str) and not pattern.match(element):
                     raise ValueError(f"Invalid {{name}} format: {element}")
         elif isinstance(v,str):
             if not pattern.match(v):
                 raise ValueError(f"Invalid {{name}} format: {v}")
-        return v
\ No newline at end of file
+        return v
```

Comment:
it's not really a problem since i figure most people will be using black formatting if given the option, so don't worry about changing it, but ffr but the absence of newline at the end of template is intentional since it's a template <3